packages/api/apps/app.mts

```diff
@@ -57,9 +56,10 @@ export async function createAppWithAi(data: CreateAppWithAiSchemaType): Promise<
   const packagesToInstall = getPackagesToInstall(plan);
 
   if (packagesToInstall.length > 0) {
+    await waitForProcessToComplete(firstNpmInstallProcess);
+
     console.log('installing packages', packagesToInstall);
-    npmInstall({
-      cwd: pathToApp(app.externalId),
+    npmInstall(app.externalId, {
```

Comment:
I introduced this `waitForProcessToComplete` function here to ensure that these npm installs during an ai app creation always happen sequentially. In practice, almost always this function should be a noop.

Reply:
Instead of `waitForProcessToComplete`, a nicer API might be to have a wrapper around `npmInstall` that returns a promise so that callers like these can just await the install call.

Reply:
+1 we never want two concurrent ones, so that guard can be abstracted away